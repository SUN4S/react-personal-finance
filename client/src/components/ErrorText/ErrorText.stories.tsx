import { Meta, Story } from "@storybook/react/types-6-0";

import { ErrorText } from "./ErrorText";

interface TemplateProps {
  theme: string;
  title: string;
}

export default {
  title: "Components/Error Text",
  component: ErrorText,
} as Meta;

const Template: Story<TemplateProps> = (args) => {
  return (
    <html data-theme={args.theme}>
      <body
        style={{
          width: "400px",
          backgroundColor: "var(--sidebar-bg)",
          padding: "8px",
        }}
      >
        <ErrorText title={args.title} />
      </body>
    </html>
  );
};

export const Default_Light = Template.bind({});
Default_Light.args = {
  theme: "light",
  title: "No Data",
};

export const Default_Dark = Template.bind({});
Default_Dark.args = {
  theme: "dark",
  title: "No Data",
};
