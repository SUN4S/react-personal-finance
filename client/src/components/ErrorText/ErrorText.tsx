import "./ErrorText.scss";

// simple function to render string
export const ErrorText = (props: { title: string }) => {
  return (
    <div className="errorText" data-testid="errorText">
      <span>{props.title}</span>
    </div>
  );
};
