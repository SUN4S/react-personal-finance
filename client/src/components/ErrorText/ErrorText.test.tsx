import { cleanup, fireEvent, render } from "@testing-library/react";

import { ErrorText } from "./ErrorText";
import renderer from "react-test-renderer";

describe("Error Text", () => {
  afterEach(cleanup);

  it("Renders", () => {
    const { getByTestId } = render(<ErrorText title="Error" />);
    expect(getByTestId("errorText")).toBeTruthy();
  });

  it("Renders Text", () => {
    const { getByTestId } = render(<ErrorText title="Error" />);
    expect(getByTestId("errorText").querySelector("span")).toHaveTextContent(
      "Error"
    );
  });

  it("Matches Snapshot", () => {
    const tree = renderer.create(<ErrorText title="Error" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
