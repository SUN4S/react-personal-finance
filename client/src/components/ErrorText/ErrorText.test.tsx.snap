// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Text Matches Snapshot 1`] = `
<div
  className="errorText"
  data-testid="errorText"
>
  <span>
    Error
  </span>
</div>
`;
